code/modules/mob/living/simple_animal/revenant/revenant_blight.dm

```diff
@@ -0,0 +1,68 @@
+/datum/disease/revblight
+	name = "Unnatural Wasting"
+	max_stages = 5
+	stage_prob = 10
+	spread_flags = NON_CONTAGIOUS
+	cure_text = "Holy water or extensive rest."
+	spread_text = "A burst of unholy energy"
+	cures = list("holywater")
+	cure_chance = 50 //higher chance to cure, because revenants are assholes
+	agent = "Unholy Forces"
+	viable_mobtypes = list(/mob/living/carbon/human)
+	disease_flags = CURABLE
+	permeability_mod = 1
+	severity = BIOHAZARD
+	var/stagedamage = 0 //Highest stage reached.
+
+/datum/disease/revblight/stage_act()
+	if(affected_mob.lying && prob(stage*2))
+		affected_mob << "<span class='notice'>You feel better.</span>"
+		cure()
+		return
+	if(prob(stage*3))
+		affected_mob << "<span class='revennotice'>You suddenly feel [pick("sick and tired", "disoriented", "tired and confused", "nauseated", "faint", "dizzy")]...</span>"
+		affected_mob.confused += 10
+		affected_mob.adjustStaminaLoss(10)
+		new/obj/effect/overlay/temp/revenant(affected_mob.loc)
+	if(stagedamage < stage)
+		stagedamage ++
```

Comment:
while I may appear to be a coder I am actually a clown wearing a keyboard as a hat